### PR TITLE
ACCUMULO-4162 Fix zookeeper version matching

### PR DESCRIPTION
**Pull request by ctubbsii**
In the course of fixing ACCUMULO-4162, I noticed lots of other small script problems which were just as likely to mess up users. So, I ran shellcheck on all the scripts in assemble/bin, and a little bit outside there. This is the results for the 1.6 branch, and could use a bit of a review/manual testing to make sure they work as expected.

Any help checking these out would be greatly appreciated.

### CODE REVIEW

File: assemble/bin/start-all.sh [hunk 1]
```diff
@@ -28,50 +28,52 @@ bin="$( cd -P "$( dirname "$SOURCE" )" && pwd )"
...
 
-for gc in `egrep -v '(^#|^\s*$)' "$ACCUMULO_CONF_DIR/gc"`; do
-   ${bin}/start-server.sh $gc gc "garbage collector"
-done
+startServersFromHostsFile() {
+  # use hostfile in conf dir to get hosts, and start each server with the remaining args
+  local hostfile; hostfile="$1"
+  shift
+  local otherArgs; otherArgs=("$@")
+  while IFS=$' \t\n' read -r host; do
```
Comment (joshelser) [inline assemble/bin/start-all.sh]: IFS=$' \t\n' is splitting on space, tab and newline?

File: assemble/bin/start-all.sh [hunk 2]
```diff
@@ -28,50 +28,52 @@ bin="$( cd -P "$( dirname "$SOURCE" )" && pwd )"
...
+    "${bin}/start-server.sh" "$host" "${otherArgs[@]}"
+  done < <(egrep -v '^(\s*#.*|\s*)$' "$ACCUMULO_CONF_DIR/$hostfile")
```
Comment (joshelser) [inline assemble/bin/start-all.sh]: The egrep pattern is slightly different here. Any chance for a comment? :)

Comment (ctubbsii) [inline assemble/bin/start-all.sh]: I just thought this version was a bit more explicit. (It's also tolerant of indented comment lines.)
Read as: lines either beginning by any amount of whitespace followed by a comment OR any amount of whitespace before termination

Another variation, which might be more readable is: '^\s*(#.*)?$'
Read as: lines starting with any amount of whitespace, optionally followed by a comment, before termination

File: assemble/bin/start-all.sh (hunk 1 above)
Comment (ctubbsii) [inline assemble/bin/start-all.sh]: Yes. This is the default behavior of IFS, but I wanted to be explicit, and didn't want to have user's exported IFS interfere.

File: assemble/bin/start-all.sh (hunk 2 above)
Comment (joshelser) [inline assemble/bin/start-all.sh]: However you think is most explicit/concise in grep syntax is fine. I'd just ask for a human readable comment there as well :)

File: assemble/bin/start-all.sh (hunk 1 above)
Comment (joshelser) [inline assemble/bin/start-all.sh]: Makes sense! That has bitten me before.

**Comment by ctubbsii** [comment]
I'm going to close this for now, in favor of a more conservative change to fix the issue in ACCUMULO-4162. I may reconsider applying these changes in the master branch at a future point.

**Pull request closed without merging**